optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/ConsecutiveIntervalInfoImpl.java

```diff
@@ -0,0 +1,282 @@
+            var newBreak = new IntervalBreakImpl<>(newCluster, nextCluster, difference);
+            clusterStartSplitPointToNextBreak.put(startSplitPoint, newBreak);
+        }
+
+        // If there a cluster before this interval, add a new break
+        // between this interval and the previous cluster
+        // (this will replace the old break, if there was one)
+        var previousClusterEntry = clusterStartSplitPointToCluster.lowerEntry(startSplitPoint);
+        if (previousClusterEntry != null) {
+            var previousCluster = previousClusterEntry.getValue();
+            var difference = differenceFunction.apply(previousCluster.getEnd(), newCluster.getStart());
+            var newBreak = new IntervalBreakImpl<>(previousCluster, newCluster, difference);
+            clusterStartSplitPointToNextBreak.put(previousClusterEntry.getKey(), newBreak);
+        }
+    }
+
+    private void removeSpannedBreaksAndUpdateIntersectedBreaks(Interval<Interval_, Point_> interval,
+            IntervalClusterImpl<Interval_, Point_, Difference_> intervalCluster) {
+        var intersectedIntervalBreakMap =
+                clusterStartSplitPointToNextBreak.subMap(
+                        ObjectUtils.defaultIfNull(clusterStartSplitPointToNextBreak.floorKey(interval.getStartSplitPoint()),
+                                interval.getStartSplitPoint()),
+                        true, interval.getEndSplitPoint(), true);
+
+        if (intersectedIntervalBreakMap.isEmpty()) {
+            return;
+        }
+
+        var clusterBeforeFirstIntersectedBreak =
+                (IntervalClusterImpl<Interval_, Point_, Difference_>) (intersectedIntervalBreakMap.firstEntry().getValue()
+                        .getPreviousIntervalCluster());
+        var clusterAfterFinalIntersectedBreak =
+                (IntervalClusterImpl<Interval_, Point_, Difference_>) (intersectedIntervalBreakMap.lastEntry().getValue()
+                        .getNextIntervalCluster());
+
+        // All breaks that are not the first or last intersected breaks will
+        // be removed (as interval span them)
+        if (!interval.getStartSplitPoint()
+                .isAfter(clusterBeforeFirstIntersectedBreak.getEndSplitPoint())) {
+            if (!interval.getEndSplitPoint().isBefore(clusterAfterFinalIntersectedBreak.getStartSplitPoint())) {
+                // Case: interval spans all breaks
+                // Ex:
+                //   -----------
+                //----  ------ -----
+                intersectedIntervalBreakMap.clear();
+            } else {
+                // Case: interval span first break, but does not span the final break
+                // Ex:
+                //   -----------
+                //----  ------   -----
+                IntervalBreakImpl<Interval_, Point_, Difference_> finalBreak =
+                        intersectedIntervalBreakMap.lastEntry().getValue();
```

Comment:
Did you by any chance omit `var` here? Because it seems to me that the compiler should be able to infer this.